src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java

```diff
@@ -0,0 +1,287 @@
+      return service.create(spaceId, membership).toBlocking().first();
+    } finally {
+      membership.setSystem(sys);
+    }
+  }
+
+  /**
+   * Update the given membership instance on Contentful.
+   * <p>
+   * Please make sure that the instance provided is fetched from Contentful. Otherwise you will
+   * get an exception thrown.
+   *
+   * @param spaceId    the id of the space to be used.
+   * @param membership the membership fetched from contentful, updated by caller, to be updated.
+   * @return the updated membership.
+   * @throws IllegalArgumentException if space id is null.
+   * @throws IllegalArgumentException if membership is null.
+   * @throws IllegalArgumentException if membership id is null.
+   * @throws IllegalArgumentException if membership does not have a version attached.
+   */
+  public CMASpaceMembership update(String spaceId, CMASpaceMembership membership) {
+    assertNotNull(spaceId, "spaceId");
+    assertNotNull(membership, "membership");
+
+    final String id = getResourceIdOrThrow(membership, "membership");
+    final Integer version = getVersionOrThrow(membership, "update");
+
+    final CMASystem sys = membership.getSystem();
+    membership.setSystem(null);
+
+    try {
+      return service.update(spaceId, id, membership, version).toBlocking().first();
+    } finally {
+      membership.setSystem(sys);
+    }
+  }
+
+  /**
+   * Delete the given membership instance.
+   * <p>
+   * Please make sure that the instance provided is fetched from Contentful. Otherwise you will
+   * get an exception thrown.
+   *
+   * @param spaceId    the id of the space to be used.
+   * @param membership the membership fetched from contentful, updated by caller, to be deleted.
+   * @return an empty string on success.
+   * @throws IllegalArgumentException if space id is null.
+   * @throws IllegalArgumentException if membership id is null.
+   */
+  public int delete(String spaceId, CMASpaceMembership membership) {
+    assertNotNull(spaceId, "spaceId");
+    assertNotNull(membership, "membership");
+
+    final String id = getResourceIdOrThrow(membership, "membership");
+
+    final CMASystem sys = membership.getSystem();
+    membership.setSystem(null);
+
+    try {
+      final Response<Void> response = service.delete(spaceId, id).toBlocking().first();
+      return response.code();
+    } finally {
+      membership.setSystem(sys);
+    }
+  }
+
+  /**
+   * Handler for asynchronous requests.
+   */
+  public final class Async {
+    /**
+     * Fetch all memberships of this space, asynchronously.
+     *
+     * @param spaceId the space identifier identifying the space.
+     * @return a callback for the array fetched.
+     * @throws IllegalArgumentException if spaceId is null.
+     * @see ModuleSpaceMemberships#fetchAll(String)
+     */
+    public CMACallback<CMAArray<CMASpaceMembership>> fetchAll(
+        final String spaceId,
+        final CMACallback<CMAArray<CMASpaceMembership>> callback
+    ) {
```

Comment:
Another great spot. It is incosistent with the rest of the sdk, so I'll move them up.